erizo/src/test/rtp/QualityManagerTest.cpp

```diff
@@ -0,0 +1,174 @@
+        stats->getNode()["qualityLayers"][spatial_layer][temporal_layer]+=
+          kBaseBitrate + (kSpatialBitrateFactor * spatial_layer) + (kTemporalBitrateFactor * temporal_layer);
+      }
+    }
+    stats_clock->advanceTime(kTestRateStatIntervalSize);
+  }
+
+  uint64_t getStatForLayer(int spatial_layer, int temporal_layer) {
+    return stats->getNode()["qualityLayers"][spatial_layer][temporal_layer].value();
+  }
+
+  void clearLayer(int spatial_layer, int temporal_layer) {
+    stats->getNode()["qualityLayers"][spatial_layer].insertStat(std::to_string(temporal_layer),
+        MovingIntervalRateStat{kTestRateStatIntervalSize, kRateStatIntervals, 1., stats_clock});
+  }
+
+  void addStatToLayer(int spatial_layer, int temporal_layer, uint64_t value) {
+    stats->getNode()["qualityLayers"][spatial_layer][temporal_layer] += value;
+  }
+
+  void setSenderBitrateEstimation(uint64_t bitrate) {
+    stats->getNode()["total"].insertStat("senderBitrateEstimation", CumulativeStat{bitrate});
+  }
+
+  void advanceClock(duration time) {
+    simulated_clock->advanceTime(time);
+  }
+
+  void advanceClockMs(uint64_t time_ms) {
+    simulated_clock->advanceTime(std::chrono::milliseconds(time_ms));
+  }
+
+  std::shared_ptr<QualityManager> quality_manager;
+  std::shared_ptr<SimulatedClock> stats_clock;
+};
+
+TEST_F(QualityManagerTest, shouldStartAtBaseLayer) {
+  EXPECT_EQ(quality_manager->getSpatialLayer() , kBaseSpatialLayer);
+  EXPECT_EQ(quality_manager->getTemporalLayer() , kBaseTemporalLayer);
+}
+
+TEST_F(QualityManagerTest, shouldChangeToHighestLayerBelowEstimatedBitrate) {
+  quality_manager->notifyQualityUpdate();
+  const int kArbitrarySpatialLayer = 1;
+  const int kArbitraryTemporalLayer = 1;
+  advanceClock(erizo::QualityManager::kMinLayerSwitchInterval + std::chrono::milliseconds(1));
+
+  setSenderBitrateEstimation(getStatForLayer(kArbitrarySpatialLayer, kArbitraryTemporalLayer) + 1);
+  quality_manager->notifyQualityUpdate();
+
+  EXPECT_EQ(quality_manager->getSpatialLayer() , kArbitrarySpatialLayer);
+  EXPECT_EQ(quality_manager->getTemporalLayer() , kArbitraryTemporalLayer);
+}
+
+TEST_F(QualityManagerTest, shouldSwitchLayerImmediatelyWhenEstimatedBitrateDescends) {
+  const int kArbitrarySpatialLayer = 1;
+  const int kArbitraryTemporalLayer = 1;
+  setSenderBitrateEstimation(getStatForLayer(kArbitrarySpatialLayer, kArbitraryTemporalLayer) + 1);
+
+  quality_manager->notifyQualityUpdate();
+
+  quality_manager->setTemporalLayer(kArbitraryTemporalLayer);
+  quality_manager->setSpatialLayer(kArbitrarySpatialLayer);
+
+  const int kArbitraryLowerSpatialLayer = 0;
+  const int kArbitraryLowerTemporalLayer = 1;
+
+  setSenderBitrateEstimation(getStatForLayer(kArbitraryLowerSpatialLayer, kArbitraryLowerTemporalLayer) + 1);
+  quality_manager->notifyQualityUpdate();
+
+  EXPECT_EQ(quality_manager->getSpatialLayer() , kArbitraryLowerSpatialLayer);
+  EXPECT_EQ(quality_manager->getTemporalLayer() , kArbitraryLowerTemporalLayer);
+}
+
```

Comment:
I'd add a test like `shouldSwitchLayerImmediatelyWhenCurrentLayerBitrateAscends` if you implement  what I mention above. We can discuss it offline anyway.